From 5f3f4b3b38bb5d4ede5f9b131f6e100d8a57a693 Mon Sep 17 00:00:00 2001
From: Kevin Wang <kevin1.wang@amd.com>
Date: Tue, 11 Jun 2019 15:21:06 +0800
Subject: [PATCH 2589/2940] drm/amd/powerplay: print smu versions only if
 version mismatch

only printf smu version when if version not matched.

Change-Id: I3b83af99a576a3bd8c4ea66227fa2120873e18ea
Signed-off-by: Kevin Wang <kevin1.wang@amd.com>
Reviewed-by: Xiaojie Yuan <xiaojie.yuan@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/powerplay/smu_v11_0.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/powerplay/smu_v11_0.c b/drivers/gpu/drm/amd/powerplay/smu_v11_0.c
index 631ee8460e38..bc396908e7b3 100644
--- a/drivers/gpu/drm/amd/powerplay/smu_v11_0.c
+++ b/drivers/gpu/drm/amd/powerplay/smu_v11_0.c
@@ -257,12 +257,12 @@ static int smu_v11_0_check_fw_version(struct smu_context *smu)
 	smu_minor = (smu_version >> 8) & 0xff;
 	smu_debug = (smu_version >> 0) & 0xff;
 
-	pr_info("SMU Driver IF Version = 0x%08x, SMU FW IF Version = 0x%08x,"
-		" SMU FW Version = 0x%08x (%d.%d.%d)\n",
-		if_version, smu->smc_if_version,
-		smu_version, smu_major, smu_minor, smu_debug);
 
 	if (if_version != smu->smc_if_version) {
+		pr_info("smu driver if version = 0x%08x, smu fw if version = 0x%08x, "
+			"smu fw version = 0x%08x (%d.%d.%d)\n",
+			smu->smc_if_version, if_version,
+			smu_version, smu_major, smu_minor, smu_debug);
 		pr_err("SMU driver if version not matched\n");
 		ret = -EINVAL;
 	}
-- 
2.17.1

